ggshield/cmd/auth/utils.py

```diff
@@ -21,7 +22,11 @@ def check_instance_has_enabled_flow(config: Config) -> None:
 
     # don't use gitguardian client because the request should not include the token
     session = create_session(config.allow_self_signed)
-    response = session.get(urljoin(config.api_url, "/v1/metadata"), timeout=30)
+
+    try:
+        response = session.get(urljoin(config.api_url, "/v1/metadata"), timeout=30)
+    except ConnectionError:
+        raise click.ClickException("Invalid GitGuardian instance url.")
```

Comment:
This looks good, but I wonder if it should not be caught upper: are we sure this call is always going to be the first API call?

Reply:
for the login command yes, but you're right, I should handle logout as well

Reply:
In fact it's already taken care of for logout:
```python
    try:
        response = client.post(endpoint="token/revoke")
    except ConnectionError:
        raise LogoutError(REVOKE_FAIL_MESSAGE)
```